### DISCUSSION TITLE
How get details about esx host?

### DISCUSSION THREAD

**Discussion by jancovic** [vmware/pyvmomi-community-samples#741] [General]
How can I get details about esx host via vcenter api python?

E.g. ram, cpu, server model?
I haven't found this info also in rest API docu. Only via powecli.
So is it even possible?

**Comment by karaatanassov** [comment]
Yes it is possible. You would do this via Pyvmomi easily.

If you have a handle to the host object already it is is literally one line. For the high level stuff [the summary property](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.host.Summary.html) should suffice
```python
summary = host.summary
print(f"memory size: {summary.hardware.memorySize}")
```

You can find interesting info in the [config](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.host.ConfigInfo.html) and [runtime](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.host.RuntimeInfo.html).

If you do not have the host system object you can find it using the [SearchIndex](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.SearchIndex.html) or [PropertyCollector](https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vmodl.query.PropertyCollector.html) APIs.

You can find some sample code w.r.t PropertyCollector. There are utilities in this project that take care of resource management [here](https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/tools/pchelper.py)

Also see [the programming guide](https://developer.vmware.com/docs/19250/GUID-F7C336F5-BBD1-4716-8B1F-026C6BBBCAEE.html)
